### PR TITLE
REST and GraohQL intro page

### PR DESCRIPTION
**Pull request by ecarson**

### CODE REVIEW

**Review by MichaelBaj — DISMISSED** [review]

**Review by Chr1st0ph3rTurn3r — APPROVED** [review]

File: docs/intro_rest_graphql_apis.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+---
+title: Introduction to REST and GraphQL APIs
+sidebar_label: REST/GraphQL
```
Comment (Chr1st0ph3rTurn3r) [inline docs/intro_rest_graphql_apis.md:3]:
```suggestion
sidebar_label: REST and GraphQL APIs
```

File: docs/intro_rest_graphql_apis.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+---
+
+The SSR provides Application Programming Interfaces (APIs) that can be used to interact with the SSR from an external application. The two primary interfaces are REST and GraphQL.
+
+A REST API (Representational State Transfer Application Programming Interface) is a set of rules and conventions for building and interacting with web services. It uses standard HTTP methods such as GET, POST, PUT, DELETE, etc., to perform operations on resources, which are typically represented in formats like JSON or XML.
+
+The GraphQL API is an alternative to REST and allows clients to request exactly the data they need using a query and the server will populate the items in the query.
```
Comment (Chr1st0ph3rTurn3r) [inline docs/intro_rest_graphql_apis.md:10]:
```suggestion
The GraphQL API is an alternative to REST and allows clients to use queries to request exactly the data they need. The server then populates the items in the query.
```

File: docs/intro_rest_graphql_apis.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+Note: The examples shown in this document use the curl command-line application, any HTTP client can be used.
```
Comment (Chr1st0ph3rTurn3r) [inline docs/intro_rest_graphql_apis.md:12]:
```suggestion
:::note
The examples shown in this document use the `curl` command-line application; any HTTP client can be used.
:::
```

File: docs/intro_rest_graphql_apis.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+## Authentication Tokens
+
+The REST and GraphQL APIs are authorized and authenticated securely using authorization tokens that are granted using the username and password of a suitably priviliged user. These tokens are passed to each API call and the SSR will use this token to determine authorization for each API call. The RBAC privileges of the user will determine access to the resources being accessed by the API.
```
Comment (Chr1st0ph3rTurn3r) [inline docs/intro_rest_graphql_apis.md:16]:
```suggestion
The REST and GraphQL APIs are authorized and authenticated securely using authorization tokens. Tokens are granted using the username and password of a suitably priviliged user, and passed to each API call. The SSR uses the token to determine authorization for each API call. The RBAC privileges of the user determine access to the resources being accessed by the API.
```

File: docs/intro_rest_graphql_apis.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+The `/api/v1/login` REST API is used to generate these tokens. For example:
+`curl --request POST -k --url 'https://192.168.0.1/api/v1/login' -H "Content-Type: application/json" -d '{ "username": "admin", "password":"128Tadmin"}'`
+
+In this example the address of the SSR is `192.168.0.1` and the username is `admin` with password `128Tadmin`. Additionally the `-H "Content-Type: application/json"` specifies
```
Comment (Chr1st0ph3rTurn3r) [inline docs/intro_rest_graphql_apis.md:21]:
```suggestion
In this example, the address of the SSR is `192.168.0.1` and the username is `admin` with password `128Tadmin`. Additionally the `-H "Content-Type: application/json"` specifies a `Content-Type` header that the client is passing and accepting JSON data.
```

File: docs/intro_rest_graphql_apis.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+a `Content-Type` header that the client is passing and accepting JSON data.
```
Comment (Chr1st0ph3rTurn3r) [inline docs/intro_rest_graphql_apis.md:22]: remove this line, it is part of the change above.

File: docs/intro_rest_graphql_apis.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+If the login attempt is successful a token will be returned, for example:
```
Comment (Chr1st0ph3rTurn3r) [inline docs/intro_rest_graphql_apis.md:24]:
```suggestion
If the login attempt is successful a token is returned. For example:
```

File: docs/intro_rest_graphql_apis.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+```
+{"token":"eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoiYWRtaW4iLCJyb2xlcyI6WyJhZG1pbiJdLCJzY29wZXMiOlsiY29uZmlndXJlIiwic2hvdy1jb21tYW5kcyJdLCJjYXBhYmlsaXRpZXMiOlsiY29uZmlnLXJlYWQiLCJjb25maWctd3JpdGUiLCJwcm92aXNpb25pbmciXSwiYXBwbGljYXRpb24iOiJ1bmtub3duIiwidXNlckFnZW50IjoiY3VybC83LjI5LjAiLCJpcEFkZHJlc3MiOiIxMjcuMC4wLjEiLCJpc3MiOiJSVFJfRUFTVF9DT05EVUNUT1IiLCJpYXQiOjE3MjY1NDQ5Mzd9.NoEgcSzm752k1PWsvi5WtyFVCA825WI_fFMfOVeoNXvK1jsyW6UKiwGD8gSJFuQrtNYISgZWlBrqD3bhpiii33-DnAzOOEIuDXpbNGKAw2KwiuVKHoDIj8iWRi1grBERFpDFKCgjO15sR0q2JAb88k_EIkIHLeuS1bLSpi1mGfjRGeNcDh8DkCjQM1jH-DbPXf5oJ7pAq79pflLR-yS5WcWpeeQRaO_xrwWnd9cS4R31T-T0p1q0SYJanB9IQ3YUtue3zqArJmb4qHT46HJ_rctpp6NLXUih2Q7LEe7-DQB3yV9nDoB5vIAZn1Ppdo2kW3_A0B-fcyaFdH_KZxkuQQ"}
```
Comment (Chr1st0ph3rTurn3r) [inline docs/intro_rest_graphql_apis.md:26]:
```suggestion
{"token":"eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoiYWRtaW4iLCJyb2xlcyI6WyJhZG1pbiJdLCJzY29wZXMiOlsiY29uZmlndXJlIiwic2hvdy1jb21tYW5kcyJdLCJjYXBhYmlsaXRpZXMiOlsiY29uZmlnLXJlYWQiLCJjb25maWctd3JpdGUiLCJwcm92aXNpb25pbmciXSwiYXBwbGljYXRpb24iOiJ1bmtub3duIiwidXNlckFnZW50IjoiThiSaintAR31lt0k3njI5LjAlc3MiOiIxMjcuMC4wLjEiLCJpc3MiOiJSVFJfRUFTVF9DT05EVUNUT1IiLCJpYXQiOjE3MjY1NDQ5Mzd9.NoEgcSzm752k1PWsvi5WtyFVCA825WI_fFMfOVeoNXvK1jsyW6UKiwGD8gSJFuQrtNYISgZWlBrqD3bhpiii33-DnAzOOEIuDXpbNGKAw2KwiuVKHoDIj8iWRi1grBERFpDFKCgjO15sR0q2JAb88k_EIkIHLeuS1bLSpi1mGfjRGeNcDh8DkCjQM1jH-DbPXf5oJ7pAq79pflLR-yS5WcWpeeQRaO_xrwWnd9cS4R31T-T0p1q0SYJanB9IQ3YUtue3zqArJmb4qHT46HJ_rctpp6NLXUih2Q7LEe7-DQB3yV9nDoB5vIAZn1PThiSaintAR31lt0k3ndH_KZxkuQQ"}
```

File: docs/intro_rest_graphql_apis.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+```
+
+This token can be passed via an `Authorization` header to subsequent API calls for authorization. For example:
+`curl --request GET -k --url 'https://192.168.0.1/api/v1/version' -H "Content-Type: application/json" -H 'Authorization: Bearer <token>`
+
+In this example the address of the SSR is `192.168.0.1`, the API being called is `/api/v1/version`, and an `Authorization` header is specifying a `Bearer` token. The value returned
```
Comment (Chr1st0ph3rTurn3r) [inline docs/intro_rest_graphql_apis.md:32]:
```suggestion
In this example, the address of the SSR is `192.168.0.1`, the API being called is `/api/v1/version`, and an `Authorization` header is specifying a `Bearer` token. The value returned from the `/api/v1/login` API should replace `<token>`. For example:
```

File: docs/intro_rest_graphql_apis.md [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+from the `/api/v1/login` API should replace `<token>`, for example:
```
Comment (Chr1st0ph3rTurn3r) [inline docs/intro_rest_graphql_apis.md:33]: remove this line, it is integrated into the change above

File: docs/intro_rest_graphql_apis.md [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+```
+curl --request GET -k --url 'https://192.168.0.1/api/v1/version' -H "Content-Type: application/json" -H 'Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoiYWRtaW4iLCJyb2xlcyI6WyJhZG1pbiJdLCJzY29wZXMiOlsiY29uZmlndXJlIiwic2hvdy1jb21tYW5kcyJdLCJjYXBhYmlsaXRpZXMiOlsiY29uZmlnLXJlYWQiLCJjb25maWctd3JpdGUiLCJwcm92aXNpb25pbmciXSwiYXBwbGljYXRpb24iOiJ1bmtub3duIiwidXNlckFnZW50IjoiY3VybC83LjI5LjAiLCJpcEFkZHJlc3MiOiIxMjcuMC4wLjEiLCJpc3MiOiJSVFJfRUFTVF9DT05EVUNUT1IiLCJpYXQiOjE3MjY1NDQ5Mzd9.NoEgcSzm752k1PWsvi5WtyFVCA825WI_fFMfOVeoNXvK1jsyW6UKiwGD8gSJFuQrtNYISgZWlBrqD3bhpiii33-DnAzOOEIuDXpbNGKAw2KwiuVKHoDIj8iWRi1grBERFpDFKCgjO15sR0q2JAb88k_EIkIHLeuS1bLSpi1mGfjRGeNcDh8DkCjQM1jH-DbPXf5oJ7pAq79pflLR-yS5WcWpeeQRaO_xrwWnd9cS4R31T-T0p1q0SYJanB9IQ3YUtue3zqArJmb4qHT46HJ_rctpp6NLXUih2Q7LEe7-DQB3yV9nDoB5vIAZn1Ppdo2kW3_A0B-fcyaFdH_KZxkuQQ'
```
Comment (Chr1st0ph3rTurn3r) [inline docs/intro_rest_graphql_apis.md:35]:
```suggestion
curl --request GET -k --url 'https://192.168.0.1/api/v1/version' -H "Content-Type: application/json" -H 'Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoiYWRtaW4iLCJyb2xlcyI6WyJhZG1pbiJdLCJzY29wZXMiOlsiY29uZmlndXJlIiwic2hvdy1jb21tYW5kcyJdLCJjYXBhYmlsaXRpZXMiOlsiY29uZmlnLXJlYWQiLCJjb25maWctd3JpdGUiLCJwcm92aXNpb25pbmciXSwiYXBwbGljYXRpb24iOiJ1bmtub3duIiwidXNlckFnZW50IjoiY3VybC83LjIThiSaintAR31lt0k3nlc3MiOiIxMjcuMC4wLjEiLCJpc3MiOiJSVFJfRUFTVF9DT05EVUNUT1IiLCJpYXQiOjE3MjY1NDQ5Mzd9.NoEgcSzm752k1PWsvi5WtyFVCA825WI_fFMfOVeoNXvK1jsyW6UKiwGD8gSJFuQrtNYISgZWlBrqD3bhpiii33-DnAzOOEIuDXpbNGKAw2KwiuVKHoDIj8iWRi1grBERFpDFKCgjO15sR0q2JAb88k_EIkIHLeuS1bLSpi1mGfjRGeNcDh8DkCjQM1jH-DbPXf5oJ7pAq79pflLR-yS5WcWpeeQRaO_xrwWnd9cS4R31T-T0p1q0SYJanB9IQ3YUtue3zqArJmb4qHT46HJ_rctpp6NLXUih2Q7LEe7-DQB3yV9nDoB5vIAZn1PThiSaintAR31lt0k3ndH_KZxkuQQ'
```

File: docs/intro_rest_graphql_apis.md [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+```
+
+## Built In Product Documentation
+
+There are numerous APIs in the SSR and these are documented in the GUI of the SSR. This is available from the `About This System` page in the GUI.
```
Comment (Chr1st0ph3rTurn3r) [inline docs/intro_rest_graphql_apis.md:40]:
```suggestion
There are numerous APIs in the SSR. The API documentation is available from the `About This System` page in the GUI.
```

File: docs/intro_rest_graphql_apis.md [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+
+There are separate documentation pages for REST and GraphQL. The documentation pages feature interactive elements that allow the APIs to be dynamically tested from the GUI.
+
+The REST APIs are documented using Swagger and can be found here:
```
Comment (Chr1st0ph3rTurn3r) [inline docs/intro_rest_graphql_apis.md:44]:
```suggestion
The REST APIs are documented using Swagger and are found here:
```

File: docs/intro_rest_graphql_apis.md [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+`https://<SSR address>/documentation/swagger`
+
+The GraphQL APIs are documented using an interactive explorer and can be found here:
```
Comment (Chr1st0ph3rTurn3r) [inline docs/intro_rest_graphql_apis.md:47]:
```suggestion
The GraphQL APIs are documented using an interactive explorer and are found here:
```

**Review by Chr1st0ph3rTurn3r — CHANGES_REQUESTED** [review]

**Review by Chr1st0ph3rTurn3r — APPROVED** [review]

**Pull request merged**